tool/yubico-piv-tool.c

```diff
@@ -724,9 +724,7 @@ static bool request_certificate(ykpiv_state *state, enum enum_key_format key_for
   }
 
   memcpy(digest, oid, oid_len);
-  /* XXX: this should probably use X509_REQ_digest() but that's buggy */
-  if(!ASN1_item_digest(ASN1_ITEM_rptr(X509_REQ_INFO), md, req->req_info,
-              digest + oid_len, &digest_len)) {
+  if (!X509_REQ_digest(req, md, digest + oid_len, &digest_len)) {
```

Comment:
Have you tested if this works? The problem with  X509_REQ_digest() is (used to be?) that it hashes the entire X509_REQ, ideally I guess there would be a X509_REQ_INFO_digest().